js-foundations.js

```diff
@@ -32,19 +94,57 @@ var testString2 = "academy"
 var testString3 = "sandiego"
 
 // 4a. Write a function called alphabetSoup that takes one argument and returns a string with all the letters in alphabetical order. Use each of the varibales as test cases. Expected output: "aelnr" "aacdemy" "adeginos"
+var testString1 = "learn"
+var testString2 = "academy"
+var testString3 = "sandiego" 
+function AlphabetSoup(str) { 
+  newArr = str.match(/[a-z]/g);
+  console.log(newArr);
+  return newArr.sort().join("");
+}
 
+ AlphabetSoup(testString1, testString2, testString3); 
 
 
 // 4b. Write a function that takes in all three variables and returns a string with all the letters in alphabetical order. Expected output: "aaaacddeeegilmnnorsy"
 
-
+var testString1 = "learn"
+var testString2 = "academy"
+var testString3 = "sandiego"
+function alphabet_order(str)
+  {
+return str.split('').sort().join('');
+  }
+console.log(alphabet_order(testString1));
+console.log(alphabet_order(testString2));
+console.log(alphabet_order(testString3));
 
 // ------------------------------------- Consider the variables:
 var amounts = [9, 1, 8, 16, 5, 1, 42]
 var animals = ["ducks", "elephants", "pangolins", "zebras", "giraffes", "penguins", "llamas"]
 
 // 5a. Write a function that takes the two variables and returns a combined string. Expected output: "9 ducks 1 elephants 8 pangolins 16 zebras 5 giraffes 1 penguins 42 llamas"
+function Container(){
+   this.variables = [];
+}
+Container.prototype.addVar = function(animals){
+   this.variables.push(animals);
+}
+Container.prototype.toString = function(){
+   var result = '';
+   for(var i in this.variables)
+       result += this.variables[i];
+   return result;
+}
+var amounts = [9, 1, 8, 16, 5, 1, 42]
+var animals = ["ducks", "elephants", "pangolins", "zebras", "giraffes", "penguins", "llamas"]
 
+var container = new Container();
+container.addVar(amounts);
+container.addVar(animals);
+container.toString();
 
 
```

Comment:
This is a really interesting approach. I like the idea of creating a class. I think it would still need a method to loop through the arrays.